### PR TITLE
Gulp setup

### PR DESCRIPTION
**Pull request by jselt**
Pull

Run `npm install`
Run `bower install`
run `gulp` 

Theme assets should be built in ./dist/

The front end won't really look any different yet :)

Running `gulp production` will minify the css and minify and uglify the js btw.

If node is installed on the linode server we could remove dist from the repo and build there - maybe use this?

https://github.com/therealklanni/git-guppy

### CODE REVIEW

**Comment by jselt** [comment]
Actually that git-guppy thing won't work.

Something more like this - http://losstopschade.de/post/96967373358

**Comment by jselt** [comment]
Ok - added some stuff, pull revert, clear cache. header should look fairly reasonable on desktop (tho not complete) and should look organized for mobile (e.g. collapses in almost right order - probably should move or hide the social icons)

File: sites/all/modules/custom/pc_block/pc_block.module [hunk 1]
```diff
@@ -81,3 +81,54 @@ function _pc_block_social_buttons() {
...
+  //setting some block titles
+  $block = $variables['block'];
+  if($block->delta == 'pc_block-secondary-nav' || $block->delta == 'pc_block-main-nav') {
+    $block->subject = '';
+    $variables['classes_array'][] = drupal_html_class($block->delta);
+  }
+  if($block->delta == 'mc_form' && $block->region == 'navigation') {
+    $block->subject = '<i class="fa fa-envelope" aria-hidden="true"></i> Sign Up';
+  }
+  //this one would not work with $block-subject - could not figure out why (normally works fine IME)
```
Comment (xaqrox) [inline sites/all/modules/custom/pc_block/pc_block.module]: I seem to remember bootstrap does something weird with this.

Comment (jselt) [inline sites/all/modules/custom/pc_block/pc_block.module]: I looked at that - it still should have worked (and has previously) - but ultimately it doesn't matter too much

Comment (jselt) [inline sites/all/modules/custom/pc_block/pc_block.module]: https://github.com/radcampaign/pulitzer2016/blob/redesign/sites/all/themes/bootstrap/templates/block/block.vars.php#L33 - is that what you were thinking of?

File: sites/all/themes/pulitzer2016/.bowerrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (xaqrox) [inline sites/all/themes/pulitzer2016/.bowerrc]: I don't really know what bower does...? But do we need to have allll this stuff in the repo? Or if we do builds on the dev server, for example, could we get away without all this?

File: sites/all/themes/pulitzer2016/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+var changed      = require('gulp-changed');
+var concat       = require('gulp-concat');
+var cleanCSS     = require('gulp-clean-css');
+var flatten      = require('gulp-flatten');
+var flexibility  = require('postcss-flexibility');
+var gulp         = require('gulp');
+var gulpif       = require('gulp-if');
+var imagemin     = require('gulp-imagemin');
+var jshint       = require('gulp-jshint');
+var lazypipe     = require('lazypipe');
```
Comment (xaqrox) [inline sites/all/themes/pulitzer2016/gulpfile.js]: This is missing from `package.json`

Comment (jselt) [inline sites/all/themes/pulitzer2016/gulpfile.js]: actually I decided not to use lazypipe and removed it from package - but forgot to remove the require. Removed now

**Pull request merged**